util/project_allocation_mapper.py

```diff
@@ -129,19 +129,22 @@ def get_all_projects(self):
                 projects[proj['chargeCode']] = proj
             projects[proj['chargeCode']]['allocations'].append(self.portal_to_tas_alloc_obj(alloc))
         return sorted(list(projects.values()), reverse=True, key=self.sort_by_allocation_request_date)
```

Comment:
Looking at this code... I have no idea how this is currently working. This will return the same key for all projects :/